### PR TITLE
Add success and failure reference endpoints to sandbox 

### PR DESCRIPTION
**Pull request by elliotcm**
## Context

This allows providers to better test the references flow over the API.

## Changes proposed in this pull request

Screenshot 2023-02-06 at 10 13 04

## Guidance to review

API requests will need to be for a reference in `requested_feedback` state.

## Things to check

- [x] API release notes have been updated if necessary

### CODE REVIEW

File: app/controllers/vendor_api/references_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        reference.update!(
+          feedback_status: 'feedback_refused',
+          feedback_refused_at: Time.zone.now,
+        )
+      end
+    end
+
+  private
+
+    def sandbox_only
```
Comment (carlosmartinez) [inline app/controllers/vendor_api/references_controller.rb:28]: do we know if `HostingEnvironment.sandbox_mode?` returns true for QA and review apps? i think we want this functionality to be available on QA. perhaps the correct test here is `HostingEnvironment.sandbox_mode? || HostingEnvironment.test_environment?`, or simply `!HostingEnvironment.production?`.

**Comment by carlosmartinez** [comment]
question around sandbox mode, otherwise looks great

File: app/controllers/vendor_api/references_controller.rb (hunk 1 above)
Comment (elliotcm) [reply app/controllers/vendor_api/references_controller.rb:28]: @carlosmartinez Updated.

**Comment by elliotcm** [comment]
@carlosmartinez "non-production" isn't quite true, as it won't be on staging, load testing, etc.

**Comment by elliotcm** [comment]
> > @carlosmartinez "non-production" isn't quite true, as it won't be on staging, load testing, etc.
> 
> true. `workflow_testing_only`?

Sure

**Pull request merged**